addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/MiHomeSubdevicesHandler.java

```diff
@@ -0,0 +1,651 @@
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "Bridge status is " + bridgeStatus.toString());
+            }
+        } else {
+            logger.warn("Cann't initialize ThingHandler, bridge is missing");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, "Bridge is missing");
+        }
+
+    }
+
+    private void initializeInternal() {
+        // Bridge is ONLINE and this should not be the case, we simply double check
+        if (gatewayHandler == null || !isGatewayReady()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Bridge isn't set up correctly");
+            return;
+        }
+
+        Integer deviceID = getDeviceID();
+        if (deviceID != null) {
+            logger.debug("Device with ID {} is already paired", deviceID);
+
+            this.mihomeID = deviceID;
+
+            JsonObject data = gatewayHandler.getSubdeviceData(this.mihomeID);
+            if (data != null) {
+                updateThingProperties(data);
+
+                updateStatus(ThingStatus.ONLINE);
+
+                scheduleRegularUpdate(updateInterval);
+            } else {
+                logger.warn("Mi|Home server may be temporary unavailable or device with ID {} may be deleted.",
+                        this.mihomeID);
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                        "Missing data for paired device with ID " + this.mihomeID);
+            }
+        } else {
+            schedulePairing(gatewayHandler.getGatewayId());
+        }
+    }
+
+    private boolean isGatewayReady() {
+        return gatewayHandler.getGatewayId() > 0;
+    }
+
+    private Integer getDeviceID() {
+        Integer id = null;
+        // The thing has been once initialized and the deviceID has been persisted
+        logger.debug("Searching deviceID in the thing configuration.");
+        Configuration config = getConfig();
+        BigDecimal deviceID = (BigDecimal) config.get(MiHomeBindingConstants.PROPERTY_DEVICE_ID);
+        if (deviceID != null) {
+            id = deviceID.intValue();
+            return id;
+        }
+
+        // The thing has been created by a DiscoveryService and the deviceID is included in the properties
+        logger.debug("Searching device ID in the thing properties.");
+        Map<String, String> properties = editProperties();
+        if (properties.containsKey(MiHomeBindingConstants.PROPERTY_DEVICE_ID)) {
+            String propertyValue = properties.get(MiHomeBindingConstants.PROPERTY_DEVICE_ID);
+            try {
+                id = Integer.parseInt(propertyValue);
+            } catch (NumberFormatException e) {
+                logger.debug("Cann't parse property {} as int, value is {}", MiHomeBindingConstants.PROPERTY_DEVICE_ID,
```

Comment:
What would a caller even do with a null id? This looks like a case where we shouldn't be masking the exception anyway, although re-raising as a more appropriate exception seems sensible, rather than returning null.